### ISSUE TITLE
Should not serialize shorthand when it is partly inherit or initial

### ISSUE THREAD

**Issue by upsuper** [servo/servo#15004]
Try the following code:
```html
<!DOCTYPE html>
<div id="p"></div>
<script>
p.setAttribute('style', 'border-top-style: dotted; border-right-style: dotted; ' +
               'border-bottom-style: dotted; border-left-style: inherit');
alert(p.style.getPropertyValue('border-style'));
</script>
```

Servo outputs: `dotted dotted dotted inherit`, while it should be empty, because part of the shorthand is `inherit`. This also applies to other CSS-wide keywords `initial` and `unset`.

**Comment by upsuper** [comment]
I guess this is somehow related to #15054 but should be separate issue, since this is for serialization, while the other is for parsing.

**Comment by dbaron** [comment]
For what it's worth, Gecko has this check in [a single place](http://searchfox.org/mozilla-central/rev/ea31c4b64f34a29415a69fb768f8051495547315/layout/style/Declaration.cpp#642).

**Comment by upsuper** [comment]
This has been fixed by #15779.
